core/src/main/java/ai/timefold/solver/core/impl/bavet/common/tuple/RecordingTupleNode.java

```diff
@@ -0,0 +1,53 @@
+package ai.timefold.solver.core.impl.bavet.common.tuple;
+
+import ai.timefold.solver.core.impl.bavet.common.AbstractNode;
+import ai.timefold.solver.core.impl.bavet.common.Propagator;
+import ai.timefold.solver.core.impl.bavet.common.StaticPropagationQueue;
+import ai.timefold.solver.core.impl.bavet.common.TupleRecorder;
+
+public class RecordingTupleNode<Tuple_ extends AbstractTuple> extends AbstractNode implements TupleLifecycle<Tuple_> {
+    private final TupleRecorder<Tuple_> tupleRecorder;
+
+    public RecordingTupleNode(TupleRecorder<Tuple_> tupleRecorder) {
+        this.tupleRecorder = tupleRecorder;
+    }
+
+    public TupleRecorder<Tuple_> getTupleRecorder() {
+        return tupleRecorder;
+    }
+
+    @Override
+    public void insert(Tuple_ tuple) {
+        tupleRecorder.accept(tuple);
+    }
+
+    @Override
+    public void update(Tuple_ tuple) {
+        tupleRecorder.accept(tuple);
+    }
+
+    @Override
+    public void retract(Tuple_ tuple) {
+        // Do nothing
+    }
+
+    @Override
+    public Propagator getPropagator() {
+        return new StaticPropagationQueue<>(new TupleLifecycle<>() {
```

Comment:
So this propagates to nowhere? I'm not sure why the type even exists; it records tuples, but doesn't do anything with them.